### DISCUSSION TITLE
Route (NAT on Source)

### DISCUSSION THREAD

**Discussion by grouprcs** [strongswan/strongswan#362] [Q&A]
Hi.

I am trying to configure some routes for a remote S2S.

StrongSwan's internal IP is **192.168.1.10**
My internal network is **192.168.1.0/24**
My external IP for example is **1.1.1.1**

The remote network is **10.0.0.0/8**
The remote external IP is **2.2.2.2**

I was asked to create a NAT on the source.
The network provided to me was **10.123.456.176/28**

So I set up StrongSwan like this:

```
left=1.1.1.1
leftsubnet=10.123.456.176/28
right=2.2.2.2
rightsubnet=10.0.0.0/8
```

So far so good, the tunnel was established.
And to test it I added IP **192.168.1.177** to StrongSwan eth1.
And I managed to access an ip on the other side (**10.0.50.1**).

So any request I make I have to start from this subnet **10.123.456.176/28**
So I imagine that I need to perform a NAT

So any request on the **10.0.0.0/8** subnet has to go to the StrongSwan server, so the source IP has to be transformed to the **10.123.456.176/28** subnet, only to then travel on the VPN.

Could anyone help me with this?
On the StrongSwan machine should I do a NAT with iptables? How would it be?

On one of the **192.168.1.50** windows machines I added a
`route ADD 10.0.0.0 MASK 255.0.0.0 192.168.1.10`
Is correct?

Thanks.

**Comment by Thermi** [comment]
Use the iptables NETMAP target in *nat POSTROUTING. That dynamically sets up SNAT in conntrack for each matching packet (*nat sees the first packet of a connection that would be ctstate NEW. (You need conntrack so the kernel can later figure out to which connection related protocol messages, like ICMP error messages, relate to and NAT them correctly.)
